app/proto/reporting.proto

```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+
+package org.couchers.api.reporting;
+
+import "google/protobuf/empty.proto";
+
+import "annotations.proto";
+
+service Reporting {
+  option (auth_level) = AUTH_LEVEL_SECURE;
+
+  rpc Report(ReportReq) returns (google.protobuf.Empty) {
+    // Report content, user, or something else
+  }
+}
+
+message ReportReq {
+  // e.g. spam, inappropriate
+  string reason = 1;
+  // short description
+  string description = 2;
+  // a reference to the content being reported, this should be filled out by the UI automatically
+  string content_ref = 3;
```

Comment:
Please describe the format of the reference, "url without host and slug parts, for example "user/1234"".